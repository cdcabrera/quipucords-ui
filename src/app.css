html,
body,
#root {
  height: 100%;
}

.fadein {
  opacity:0;
  animation:fadein .15s ease-in forwards;
  animation-delay:.15s;
}

@keyframes fadein { 0%{opacity:0} 100%{opacity:1} }
