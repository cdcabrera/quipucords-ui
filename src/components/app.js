import React from 'react';
import PropTypes from 'prop-types';
import { connectRouter, reduxActions } from '../redux';
import { helpers } from '../common/helpers';
import I18n from './i18n/i18n';
import Authentication from './authentication/authentication';
import PageLayout from './pageLayout/pageLayout';
import Router from './router/router';
import ToastNotificationsList from './toastNotificationsList/toastNotificationsList';
import ConfirmationModal from './confirmationModal/confirmationModal';
import AboutModal from './aboutModal/aboutModal';
import AddSourceWizard from './addSourceWizard/addSourceWizard';
import CreateCredentialDialog from './createCredentialDialog/createCredentialDialog';
import CreateScanDialog from './createScanDialog/createScanDialog';
import MergeReportsDialog from './mergeReportsDialog/mergeReportsDialog';

class App extends React.Component {
  componentDidMount() {
    const { getLocale } = this.props;

    getLocale();
  }

  render() {
    const { locale } = this.props;

    return (
      <I18n locale={(locale && locale.value) || null}>
        <Authentication>
          <PageLayout>
            <Router />
            <ToastNotificationsList />
            <ConfirmationModal />
            <AboutModal />
            <AddSourceWizard />
            <CreateCredentialDialog />
            <CreateScanDialog />
            <MergeReportsDialog />
          </PageLayout>
        </Authentication>
      </I18n>
    );
  }
}

App.propTypes = {
  getLocale: PropTypes.func,
  locale: PropTypes.shape({
    value: PropTypes.string
  })
};

App.defaultProps = {
  getLocale: helpers.noop,
  locale: {}
};

const mapDispatchToProps = dispatch => ({
  getLocale: () => dispatch(reduxActions.user.getLocale())
});

const mapStateToProps = state => ({ locale: state.user.session.locale });

const ConnectedApp = connectRouter(mapStateToProps, mapDispatchToProps)(App);

export { ConnectedApp as default, ConnectedApp, App };
