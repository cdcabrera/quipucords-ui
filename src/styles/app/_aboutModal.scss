.quipucords-about-modal {
  // background-image: url('../images/about-bg.png');
  // background-size: 50% auto;
  // background-position: bottom right;
  // background-repeat: no-repeat;

  &.pf-c-about-modal-box {
    // background-color: var(--quipucords--gray-dark:);
  }

  .pf-c-about-modal-box__hero {
    // display: none;
  }

  .pf-c-content {
    font-size: inherit;
  }

  .quipucords-about-modal-copy-footer {
    // margin-top: -28px;
    text-align: right;
  }

  .quipucords-about-modal-copy-button {
    box-shadow: 0 0 3px var(--pf-global--palette--black-300);
  }

  .quipucords-about-modal-list {
    box-shadow: 0 0 2px var(--pf-global--palette--black-300);
    padding: 10px;

    &:focus,
    &:hover {
      box-shadow: 0 0 5px var(--pf-global--palette--black-100);
    }

    .list-unstyled li {
      display: flex;

      strong {
        white-space: nowrap;
      }
    }
  }
}
