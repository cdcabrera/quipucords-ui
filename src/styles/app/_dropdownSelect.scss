.quipucords-select {
  position: relative;

  &-pf__no-toggle-text {
    span.pf-c-select__toggle-arrow {
      margin-left: var(--pf-c-select__toggle-arrow--MarginRight);
    }

    span.pf-c-select__toggle-text {
      display: none;
    }
  }

  &-pf__position-right {
    > .pf-c-select__menu {
      right: 0;
    }
  }

  &__inline {
    display: inline-block;
  }

  &__not-inline {
    width: 100%;
  }
}

/*
.pf-c-page__header-tools-item {
  .quipucords-select {
    // &__not-inline {
    // height: 100%;

    &__not-inline {
      height: 100%;
      button.pf-c-dropdown__toggle {
        display: flex;
      }

      .pf-c-dropdown__toggle::before, .pf-c-dropdown__toggle.pf-m-action .pf-c-dropdown__toggle-button::before {

        position: absolute;
        top: 0;
        right: 0;
        bottom: 0;
        left: 0;
        content: "";
        // border-bottom: var(--pf-c-dropdown__toggle--before--BorderWidth) solid;
        // border-bottom-color: var(--pf-c-dropdown__toggle--before--BorderBottomColor);
        border: var(--pf-c-dropdown__toggle--before--BorderWidth) solid;
        // border: var(--pf-global--BorderWidth--sm) solid;
        // border-color: var(--pf-c-dropdown__toggle--before--BorderTopColor) var(--pf-c-dropdown__toggle--before--BorderRightColor) var(--pf-c-dropdown__toggle--before--BorderBottomColor) var(--pf-c-dropdown__toggle--before--BorderLeftColor);
        // border-color: var(--pf-global--BorderColor--300) var(--pf-global--BorderColor--300) var(--pf-global--BorderColor--200) var(--pf-global--BorderColor--300);
        // border-color: var(--pf-c-dropdown__toggle--before--BorderTopColor) var(--pf-global--BorderColor--200) var(--pf-c-dropdown__toggle--before--BorderBottomColor) var(--pf-c-dropdown__toggle--before--BorderRightColor);
        border-top-color: var(--pf-c-dropdown__toggle--before--BorderTopColor);
        border-left-color: var(--pf-global--Color--300);
        border-right-color: var(--pf-global--Color--300);
        border-bottom-color: var(--pf-c-dropdown__toggle--before--BorderBottomColor);
        border-bottom-width: 0;

        &:hover, &:focus, &:active {
          // border-bottom-width: var(--pf-global--BorderWidth--md);
        }
      }
    }
  }
}
 */
