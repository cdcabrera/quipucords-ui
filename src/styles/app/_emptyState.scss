.quipucords-empty-state {
  .pf-c-empty-state__primary .pf-c-button,
  .pf-c-empty-state__secondary .pf-c-button,
  .pf-c-empty-state__body {
    font-size: inherit;
  }

  &__alert {
    .pf-c-empty-state__content {
      text-align: left;
      --pf-c-empty-state__content--MaxWidth: var(--pf-c-empty-state--m-lg__content--MaxWidth);

      @media (min-width: $pf-global--breakpoint--md) {
        min-width: var(--pf-c-empty-state__content--MaxWidth);
      }
    }
  }
}
