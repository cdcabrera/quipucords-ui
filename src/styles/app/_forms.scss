/* _forms.scss */
.quipucords {
  .has-error textarea.form-control {
    border-color: $color-pf-red-100;
  }
}

textarea.vertical-scroll {
  resize: vertical;
}

.quipucords-dropdownselect {
  &.dropdown {
    width:100%;

    .dropdown-menu,
    .dropdown-toggle.btn {
      text-align: inherit;
      width: 100%;
    }

    .dropdown-toggle.btn {
      display: flex;
      flex-wrap: nowrap;
      align-items: center;

      > span:first-child {
        flex-direction: column;
        flex: 2;
        overflow: hidden;
        text-overflow: ellipsis;
        width: 1px;
      }
    }

    .quipucords-dropdownselect-menuitem {
      margin-left: 0;
      margin-right: 0;
    }

    .quipucords-dropdownselect-menuitemname {
      padding-left: 0;
      padding-right: 0;
    }

    .quipucords-dropdownselect-menuitemcheck {
      padding-left: 0;
      padding-right: 0;
      text-align: right;
    }
  }
}

.cloudmeter-touchspin {
  &.cloudmeter-scan-dialog-touchspin {
    width: 30%;
  }

  &.form-group {
    margin-left: 0;
    margin-right: 0;
  }

  .cloudmeter-touchspin-input,
  .cloudmeter-touchspin-max-button {
    left: -1px;
  }
}

.quipucords-form-control.form-control[readonly] {
  background-color: $color-pf-white;
  border: 0;
  color: inherit;
  font-weight: 700;

  &[type=textarea] {
    resize: none;
  }
}

textarea.quipucords-form-control.form-control[readonly] {
  resize: none;
}
