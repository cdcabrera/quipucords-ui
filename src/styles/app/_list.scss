.quipucords-list__overflow-scroll {
  overflow-y: auto;
  min-height: 4em;
  max-height: 12em;
}
