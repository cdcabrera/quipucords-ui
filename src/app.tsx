/**
 * Root component of the application that sets up global styles, localization, secure API requests, and routing.
 * It includes automatic token authorization for axios requests and wraps the UI with localization and routing contexts.
 *
 * @module app
 */
import React, { useEffect } from 'react';
import '@patternfly/react-core/dist/styles/base.css';
import { BrowserRouter } from 'react-router-dom';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { ReactQueryDevtools } from '@tanstack/react-query-devtools';
import axios from 'axios';
import { I18n } from './components/i18n/I18n';
import { useLocale } from './components/sessionContext/sessionProvider';
import { AppLayout } from './components/viewLayout/viewLayout';
import { AppRoutes } from './routes';
import './app.css';

const App: React.FC = () => {
  const locale = useLocale();
  const queryClient = new QueryClient();

  useEffect(() => {
    if (localStorage.getItem('authToken')) {
      axios.interceptors.request.use(
        config => ({
          ...config,
          headers: {
            ...config.headers,
            Authorization: `Token ${localStorage.getItem('authToken')}`
          } as any
        }),
        error => {
          console.error('Failed to set axios configuration', error);
        }
      );
    }
  }, []);

  return (
    <React.StrictMode>
      <QueryClientProvider client={queryClient}>
        {/* <SessionProvider> TODO: possibly add this back in when we do login/auth stuff */}
        <I18n locale={locale}>
          <BrowserRouter>
            <AppLayout>
              <AppRoutes />
            </AppLayout>
          </BrowserRouter>
        </I18n>
        {/* </SessionProvider> */}
        <ReactQueryDevtools />
      </QueryClientProvider>
    </React.StrictMode>
  );
};

export { App as default, App };
