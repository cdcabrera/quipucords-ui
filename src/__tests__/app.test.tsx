import React from 'react';
import { render } from '@testing-library/react';
// import { shallowComponent } from '../../config/jest.setupTests';
import { App } from '../app';

describe('App', () => {
  it('should render a basic component', async () => {
    const props = {};
    const { asFragment } = render(<App {...props} />);
    expect(asFragment()).toMatchSnapshot('basic');
  });
});
