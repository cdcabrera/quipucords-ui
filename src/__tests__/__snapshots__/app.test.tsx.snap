// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render a basic component: basic 1`] = `
<I18n
  locale={null}
>
  <Login>
    <BrowserRouter>
      <AppLayout>
        <AppRoutes />
      </AppLayout>
    </BrowserRouter>
  </Login>
</I18n>
`;
